src/OrchardCore/Orchard.ContentManagement/DefaultContentManager.cs

```diff
@@ -371,6 +371,17 @@ public void Create(ContentItem contentItem, VersionOptions options)
             }
         }
 
+        public void Update(ContentItem contentItem)
+        {
+            var context = new UpdateContentContext(contentItem);
+
+            Handlers.Invoke(handler => handler.Updating(context), _logger);
+            Handlers.Reverse().Invoke(handler => handler.Updated(context), _logger);
+
+            _session.Save(contentItem);
+            _contentManagerSession.Store(contentItem);
```

Comment:
Yes, you are right, it will fail. Should have checked what it does before copying...
https://github.com/OrchardCMS/Orchard2/blob/master/src/OrchardCore/Orchard.ContentManagement/DefaultContentManagerSession.cs#L18-L19